kingpin/actors/aws/iam/entities.py

```diff
@@ -525,23 +580,96 @@ def __init__(self, *args, **kwargs):
         self.get_entity_policy = self.iam_conn.get_user_policy
         self.put_entity_policy = self.iam_conn.put_user_policy
 
+        # Parse the supplied inline policies
+        self._parse_inline_policies(self.option('inline_policies'))
+
+    @gen.coroutine
+    def _ensure_groups(self, name, groups, purge):
+        """Ensure that this user is a member of specific groups.
+
+        args:
+            name: The user we're managing
+            groups: The list (or single) of groups to join be members of
+            purge: Whether or not to purge unmanaged targets.
+        """
+        if isinstance(groups, basestring):
+            groups = [groups]
+
+        current_groups = []
+        try:
+            raw = yield self.thread(self.iam_conn.get_groups_for_user, name)
+            resp_key = 'list_groups_for_user_response'
+            result_key = 'list_groups_for_user_result'
+            existing_groups = raw[resp_key][result_key]['groups']
+
+            # Quickly just pull out the friendly names
+            current_groups = [group['group_name'] for group in existing_groups]
```

Comment:
If you change this list comprehension to a set-comprehension you can get very easy math out of it. Here's a full blown example!

```python
import boto.iam
ic = boto.iam.connection.IAMConnection()
all_groups = ic.get_groups_for_user('mwise-test-user-1');
all_groups
# 
{u'list_groups_for_user_response': {u'list_groups_for_user_result': {u'groups': [{u'arn': u'arn:aws:iam::364942603424:group/mwise-test-group',
     u'create_date': u'2016-04-22T16:32:04Z',
     u'group_id': u'AGPAIECWMG5SNOZG2GPJY',
     u'group_name': u'mwise-test-group',
     u'path': u'/'},
    {u'arn': u'arn:aws:iam::364942603424:group/mwise-test-group-2',
     u'create_date': u'2016-04-22T16:32:05Z',
     u'group_id': u'AGPAJXSEU7BYYYUNE4E5W',
     u'group_name': u'mwise-test-group-2',
     u'path': u'/'}],
   u'is_truncated': u'false'},
  u'response_metadata': {u'request_id': u'a92cd353-08c7-11e6-8844-2b912bbfcb1f'}}}

current_groups = {g['group_name'] for g in all_groups['list_groups_for_user_response']
                                                     ['list_groups_for_user_result']
                                                     ['groups']}
current_groups
# {u’mwise-test-group', u'mwise-test-group-2'}

ensure_groups = {u'mwise-test-group-2', u'mwise-test-group-3'}

to_create = ensure_groups - current_groups
to_create
#  {u’mwise-test-group-3'}

to_delete = current_groups - ensure_groups
to_delete
# {u’mwise-test-group'}
```